cpp/benchmarks/io/text/multibyte_split.cpp

```diff
@@ -135,22 +135,99 @@ static void BM_multibyte_split(benchmark::State& state)
   state.counters["peak_memory_usage"] = mem_stats_logger.peak_memory_usage();
 }
 
+static void BM_multibyte_split_byte_range(benchmark::State& state)
+{
+  auto source_type        = static_cast<data_chunk_source_type>(state.range(0));
+  auto delim_size         = state.range(1);
+  auto delim_percent      = state.range(2);
+  auto file_size_approx   = state.range(3);
+  auto byte_range_percent = state.range(4);
+
+  auto byte_range_factor = static_cast<double>(byte_range_percent) / 100;
+  CUDF_EXPECTS(delim_percent >= 1, "delimiter percent must be at least 1");
+  CUDF_EXPECTS(delim_percent <= 50, "delimiter percent must be at most 50");
+  CUDF_EXPECTS(byte_range_percent >= 1, "byte range percent must be at least 1");
+  CUDF_EXPECTS(byte_range_percent <= 50, "byte range percent must be at most 50");
+
+  auto delim = std::string(":", delim_size);
```

Comment:
I believe you want this constructor
```suggestion
  auto delim = std::string(delim_size, ':');
```
The current one will try to read garbage data passed the ":".
https://godbolt.org/z/Y5Mvrzvnq

Reply:
I wonder if this would change the benchmark results since anything other the `delim_size==1` would've never found a matching delimiter, right?

Reply:
we pass the delimiter to `create_random_input`, so it *shouldn't* make a difference, but it could very well be, since the structure of the delimiter can also influence the output size (we output the location of all overlapping delimiters)